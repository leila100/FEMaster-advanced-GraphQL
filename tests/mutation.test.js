const gql = require("graphql-tag");
const createTestServer = require("./helper");

const CREATE_POST = gql`
  mutation {
    createPost(input: { message: "This is a message" }) {
      id
      message
      createdAt
      likes
      views
    }
  }
`;

describe("mutations", () => {
  test("create post", async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          createOne: jest.fn(() => {
            return { id: 1, message: "This is a message", createdAt: 12345839, likes: 20, views: 300 };
          }),
        },
      },
    });

    const res = await mutate({ mutation: CREATE_POST });
    expect(res).toMatchSnapshot();
  });
});
