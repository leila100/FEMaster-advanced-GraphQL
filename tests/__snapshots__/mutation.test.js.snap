// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations create post 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "createdAt": "12345839",
      "id": "1",
      "likes": 20,
      "message": "This is a message",
      "views": 300,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
